.github/workflows/build-with-bal-test-graalvm.yml

```diff
@@ -0,0 +1,38 @@
+name: GraalVM Check
+
+on:
+
+  workflow_dispatch:
+        inputs:
+            lang_tag:
+                description: Branch/Release Tag of the Ballerina Lang
+                required: true
+                default: main
+            lang_version:
+                description: Ballerina Lang Version (If given ballerina lang buid will be skipped)
+                required: false
+                default: ''
+            native_image_options:
+                description: Default native-image options
+                required: false
+                default: ''
+    schedule:
+        -   cron: '30 18 * * *'
+    pull_request:
+        branches: 
+            - main
+        types: [opened, synchronize, reopened, labeled, unlabeled]
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.run_id }}
+    cancel-in-progress: true
+
+jobs:
+    call_stdlib_workflow:
+        name: Run StdLib Workflow
+        if: ${{ github.event_name != 'schedule' || (github.event_name == 'schedule' && github.repository_owner == 'ballerina-platform') }}
+        uses: ballerina-platform/ballerina-standard-library/.github/workflows/build-with-bal-test-graalvm-template.yml@main
+        with:
+            lang_tag: ${{ inputs.lang_tag }}
+            lang_version: ${{ inputs.lang_version }}
+            native_image_options: '-J-Xmx7G ${{ inputs.native_image_options }}'
```

Comment:
Do we need 7Gb for edi build?

Reply:
By default, GitHub workflows only use 80% of the available memory. Currently, GH workflows are allocated with 7GB of memory, and sometimes there is a possibility of out-of-memory situations due to the memory usage of GraalVM, resulting in errors. Therefore, as a safety measure, the workflows that use the remote workflow in [bal-lib/build-with-bal-test-graalvm-template.yml](https://github.com/ballerina-platform/ballerina-library/blob/main/.github/workflows/build-with-bal-test-graalvm-template.yml) utilize the full 7GB of memory to avoid that. These are the modules that use that approach.

- https://github.com/ballerina-platform/module-ballerina-grpc/blob/master/.github/workflows/build-with-bal-test-graalvm.yml
- https://github.com/ballerina-platform/module-ballerinax-java.jdbc/blob/master/.github/workflows/build-with-bal-test-graalvm.yml
- https://github.com/ballerina-platform/module-ballerinax-postgresql/blob/main/.github/workflows/build-with-bal-test-graalvm.yml
- https://github.com/ballerina-platform/module-ballerinax-mysql/blob/master/.github/workflows/build-with-bal-test-graalvm.yml
- https://github.com/ballerina-platform/module-ballerina-graphql/blob/master/.github/workflows/build-with-bal-test-graalvm.yml
- https://github.com/ballerina-platform/module-ballerinax-kafka/blob/master/.github/workflows/build-with-bal-test-graalvm.yml
- https://github.com/ballerina-platform/module-ballerinax-mssql/blob/main/.github/workflows/build-with-bal-test-graalvm.yml